## Bayesian data analysis assignment

### Question 1

Two-thirds of sushi restaurants are legit while one-third are sketchy. Even legit sushi restaurants serve bad sushi 10% of the time while even sketchy sushi restaurants serve good sushi 50% of the time. Suppose you try a new sushi restaurant not knowing whether it is legit or sketchy. If the sushi is good then what is the probability you are in a legit sushi restaurant?

Write your answer here in table format:

| $\theta$ | $p(\theta)$ | $p(x \mid \theta)$ | $p(\theta) p(x \mid \theta)$ | $p(\theta \mid x)$  |
|----------|-------------|--------------------|------------------------------|---------------------|
| legit    | 2/3         | 9/10               | 2/3 * 9/10 = 3/5             | 3/5 / 23/30 = 18/23 |
| skechy   | 1/3         | 1/2                | 1/3 * 1/2 = 1/6              | 1/6 / 23/30 = 5/23  |

### Question 2

Solve Problem 3 from the `1_Discrete_Bayes_example` notes a different way: this time, do it sequentially by calculating the posterior probability from the first draw of a red ball and using that posterior as the prior for drawing the second red ball. (This is similar to the second solution of Problem 2 from those notes, but even easier. Since the sampling is done *with replacement*, the draws are independent, so the second draw is still out of 5 balls and the likelihood won't change from the first draw to the second.)

Your answer should be in the form of two tables like the ones in `1_Discrete_Bayes_example`, one for the first red draw and one for the second. You can copy, paste, and modify the markdown to create those tables.


Problem 3

Now suppose we draw the balls from the urn with replacement, meaning that we draw the ball, then we put it back before drawing again.

If we draw two red balls with replacement, what is the posterior probability distribution? (We’ll still assume a uniform prior.)

Write your answer here in table format: 1st draw

| $\theta$ | $p(\theta)$ | $p(x \mid \theta)$ | $p(\theta) p(x \mid \theta)$ | $p(\theta \mid x)$ |
|----------|-------------|--------------------|------------------------------|--------------------|
| H1       | 1/6         | 0                  | 0                            | 0                  |
| H2       | 1/6         | 1/5                | 1/6 * 1/5 = 1/30             | 1/30 / 1/2 = 1/15  |
| H3       | 1/6         | 2/5                | 1/6 * 2/5 = 1/15             | 1/15 / 1/2 = 2/15  |
| H4       | 1/6         | 3/5                | 1/6 * 3/5 = 1/10             | 1/10 / 1/2 = 1/5   |
| H5       | 1/6         | 4/5                | 1/6 * 4/5 = 2/15             | 2/15 / 1/2 = 4/15  |
| H6       | 1/6         | 1                  | 1/6 * 1 = 1/6                | 1/6 / 1/2 = 1/3    |

Write your answer here in table format: 2nd draw

| $\theta$ | $p(\theta)$ | $p(x \mid \theta)$ | $p(\theta) p(x \mid \theta)$ | $p(\theta \mid x)$    |
|----------|-------------|--------------------|------------------------------|-----------------------|
| H1       | 0           | 0                  | 0                            | 0                     |
| H2       | 1/15        | 1/5                | 1/15 * 1/5 = 1/75            | 1/75 / 11/15 = 1/55   |
| H3       | 2/15        | 2/5                | 2/15 * 2/5 = 4/75            | 4/75 / 11/15 = 4/55   |
| H4       | 1/5         | 3/5                | 1/5 * 3/5 = 9/75             | 9/75 / 11/15 = 9/55   |
| H5       | 4/15        | 4/5                | 4/15 * 4/5 = 16/75           | 16/75 / 11/15 = 16/55 |
| H6       | 1/3         | 1                  | 1/3 * 1 = 25/75 = 1/3        | 1/3 / 11/15 = 5/11    |

### Question 3

Suppose that we wish to predict whether a given stock will issue a dividend this year ("Yes" or "No") based on $x$, last year's profit (in some unspecified units that don't matter for this problem). We examine a large number of companies and discover that the mean value of $x$ for companies that issued a dividend was $\bar{x} = 10$, while the mean for those that didn't was $\bar{x} = 0$. In addition, the standard deviation of $x$ for these two sets of companies was $\sigma = 6$. Finally, 80% of companies issued dividends. Assuming that $x$ follows a normal distribution, **predict the probability that a company will issue a dividend this year given that its profit was** $x = 4$ last year.

We'll use Bayes's Theorem to solve this problem. The parameter $\theta$ is a discrete parameter that can be either $Y$ or $N$ (for "Yes" or "No") representing whether a stock issues a dividend or not.

Since the process is a little involved, we'll break it down into easy-to-digest chunks. Make sure you're clear on each step before moving onto the next.

#### (a)

Write down a contextually meaningful interpretation of the mathematical expression $p(\theta = Y \mid x = 4)$. (This is the posterior that we need to calculate.)

Answer:

The probability that a company will issue a dividend this year given that the profit was 4 units.

#### (b)

Write down a contextually meaningful interpretation of the prior $p(\theta = Y)$.

Answer:

The prior that a randomly chosen company will issue a dividend this year.

#### (c)

According to the information given in the problem statement, what is the value of the prior $p(\theta = Y)$? Store this value in a variable called `prior`.

In [22]:
# Store this value in a variable called prior.
prior = .8

print(prior)

0.8


#### (d)

Write down a contextually meaningful interpretation of the likelihood $p(x = 4 \mid \theta = Y)$.

Answer:

The probability that a company's profit was 4 units given that they gave out dividends.

#### (e)

To calculate the likelihood $p(x = 4 \mid \theta = Y)$, we recall that last year's profits for companies that issue dividends are distributed according to the normal model $N(10, 6)$. The general formula for a normal model is

$$\frac{1}{\sigma \sqrt{ 2 \pi}}e^{-(x - \mu)^{2}/2\sigma^{2}}$$ Plug in all the known values to obtain a value for the likelihood. In the `math` package, the Python command for the square root is `sqrt`. The number $\pi$ is stored as `pi`. Powers are computed using the caret `**` except the exponential function, which uses `exp`. For example, $e^{2\sqrt{1 + \pi^{2}}}$ would be calculated using the Python code `exp(2 * sqrt(1 + pi**2))`. Be careful with parentheses and order of operations!

Store your answer in a variable called `likelihood`. (Hint: the answer should be 0.04032845)

In [23]:
# evaluate the normal model N(10, 6) at x = 4
# and store the value in a variable called likelihood.

import math
x = 4.0
mu = 10.0
sigma = 6.0

likelihood = (1.0 / (sigma * math.sqrt(2 * math.pi))) \
             * math.exp(-((x - mu) ** 2) / (2 * sigma ** 2))

print(likelihood)

0.0403284540865239


#### (f)

The denominator of Bayes's Theorem (the "evidence") is

$$\sum_{\theta} p(\theta) p(x \mid \theta).$$

In this problem, the sum is only two terms, corresponding to the two possible values of $\theta$ ($Y$ or $N$). So it looks like the following:

$$p(\theta = Y) p(x = 4 \mid \theta = Y) + p(\theta = N) p(x = 4 \mid \theta = N).$$

You already know the first term of the sum $p(\theta = Y) p(x = 4 \mid \theta = Y)$. (This is just `prior * likelihood`.)

What is the value of $p(\theta = N)$? (You know $p(\theta = Y)$, so don't overthink this!)

Answer:

$$p(θ=N) = 1−0.8 = 0.2$$

#### (g)

To compute $p(x = 4 \mid \theta = N)$?, we'll do something very similar to what we did to compute the likelihood in Part (e). But now the normal model changes from $N(10, 6)$ to $N(0, 6)$. Why?

Answer:

$$p(x = 4 \mid \theta = N) = \frac{1}{6 \sqrt{ 2 \pi}}e^{\frac{-(4 - 0)^{2}}{2*6^{2}}}$$

#### (h)

Using the normal model $N(0, 6)$, compute $p(x = 4 \mid \theta = N)$.

In [24]:
# Evaluate the normal model N(0, 6) at x = 4.
p_no = .2
mu_no = 0.0
likelihood_no = (1/(sigma*math.sqrt(2*math.pi))) * math.exp(-((x-mu_no)**2)/(2*sigma**2))

p_no, likelihood_no

(0.2, 0.053241334253725375)

#### (i)

Put parts (f), (g), and (h) all together now to compute the denominator for Bayes's Theorem:

$$p(\theta = Y) p(x = 4 \mid \theta = Y) + p(\theta = N) p(x = 4 \mid \theta = N).$$

Store your answer in a variable called `evidence`.

In [25]:
# evaluate the denominator for Bayes's Theorem (the evidence)
# and store the value in a variable called evidence.

evidence = prior * likelihood + p_no * likelihood_no
print(evidence)

0.042911030119964194


#### (j)

If everything worked correctly, use the Bayesian theorem to calculate the posterior and explain the meaning of the posterior.

The answer should be 0.7518525. (If you didn't get that, go back and check your work *carefully*!) State this result as a conclusion in the context of the original problem.

In [26]:
answer = (prior * likelihood) / evidence
print(answer)

0.7518524532975261


Answer:

Given that the company's profit last year was 4 units, the probability that the company issued a dividend was about 75.185%